apps/web-giddh/src/app/vouchers/preview/preview.component.ts

```diff
@@ -1,90 +1,1170 @@
+    @ViewChild('paymentDialog', { static: true }) public paymentDialog: TemplateRef<any>;
+    /** Attached document preview container instance */
+    @ViewChild('attachedDocumentPreview', { static: true }) attachedDocumentPreview: ElementRef;
     /** Observable to unsubscribe all the store listeners to avoid memory leaks */
     private destroyed$: ReplaySubject<boolean> = new ReplaySubject(1);
+    /** Voucher PDF for preview downloading is in progress Observable */
+    public isVoucherDownloading$: Observable<boolean> = this.componentStore.select(state => state.isVoucherDownloading);
+    /** Voucher Download Error in PDF for preview downloading status Observable */
+    public isVoucherDownloadError$: Observable<boolean> = this.componentStore.select(state => state.isVoucherDownloadError);
+    /** Get Vouchers is in progress Observable */
+    public getVouchersInProgress$: Observable<any> = this.componentStore.getLastVouchersInProgress$;
+    /** Holds Filtered Invoice List for invoice search */
+    public filteredInvoiceList$: Subject<any[]> = new Subject<any[]>();
+    /** This will hold local JSON data */
+    public localeData: any = {};
+    /** This will hold common JSON data */
+    public commonLocaleData: any = {};
+    /** Hold day js reference */
+    public dayjs = dayjs;
+    /** Holds advance Filters keys */
+    public advanceFilters: any = {
+        page: 1,
+        count: PAGINATION_LIMIT,
+        q: '',
+        sort: '',
+        sortBy: ''
+    };
+    /** Holds search voucher form control */
+    public search: FormControl = new FormControl('');
+    /** Holds invoice list */
+    public invoiceList: any[] = [];
+    /** Holds Current selected invoice */
+    public selectedInvoice: any;
+    /** Hold invoice  type */
+    public voucherType: any = '';
+    /** Holds Total Results Count */
+    public totalPages: number = 0;
+    /** Holds params value */
+    public params: any = {};
+    /** Holds true show Payment Details enable */
+    public showPaymentDetails: boolean;
+    /** Holds true if company mode */
+    public isCompany: boolean;
+    /** Holds create new voucher text and url */
+    public createNewVoucher: any = {
+        text: '',
+        link: ''
+    };
+    /** Holds invoice type boolean status */
+    public invoiceType: any = {
+        isSalesInvoice: true,
+        isCashInvoice: false,
+        isCreditNote: false,
+        isDebitNote: false,
+        isPurchaseInvoice: false,
+        isProformaInvoice: false,
+        isEstimateInvoice: false,
+        isPurchaseOrder: false,
+        isReceiptInvoice: false,
+        isPaymentInvoice: false
+    };
+    /** Send Email Dialog Ref */
+    public sendEmailModalDialogRef: MatDialogRef<any>;
+    /** Holds Voucher Details Dialog Ref */
+    public voucherDetails: any;
+    /** Holds voucher totals */
+    public voucherTotals: any = {
+        totalAmount: 0,
+        totalDiscount: 0,
+        totalTaxableValue: 0,
```

Comment:
confirmed with @ravindersinghdang sir for using directly object without making interface now

Reply:
Please make this approved from your end if @ravindersinghdang sir told to make interface for this I will make in stage pr

Reply:
How can I approve for this?, if I raised this point.